src/utils/data_analyzer.py

```diff
@@ -0,0 +1,114 @@
+import numpy as np
+import json
+import pprint
+
+class DataAnalyzer:
+    def __init__(self, file_name):
+        self.file_name = file_name
+        self.measurements = self.load_data()
+        self.total_data = len(self.measurements)
+
+        self.outlier_data_info = {}
+        self.incomplete_data_info = {}
+        
+        self.data_with_outliers = []
+        self.data_without_outliers = []
+        self.data_incomplete = []
+        self.data_complete = []
+        self.data_complete_without_outliers = []
+        
+        self.total_complete_data_without_outliers = 0
+        self.total_outliers = 0
+        self.total_incomplete_data = 0
+
+
+    def load_data(self):
+        try:
+            with open(f'./captures/{self.file_name}', 'r') as file:
+                data = json.load(file)
+        except FileNotFoundError:
+            print("File not found. Please check the file path.")
+            return []
+        except json.JSONDecodeError as e:
+            raise Exception(f"Error decoding JSON: {e}")
+        
+        anchor_distances = []
+        for _, packet in data:
+            distances = packet["measurements"]
+            anchor_distances.append([distances.get("81", 0), distances.get("82", 0), distances.get("83", 0), distances.get("84", 0)])
```

Comment:
It's probably best to use the `Measurements` class throughout but not a requirement

Reply:
Nice to have